synapse/replication/slave/storage/transactions.py

```diff
@@ -13,19 +13,10 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-from synapse.storage import DataStore
 from synapse.storage.transactions import TransactionStore
 
 from ._base import BaseSlavedStore
 
 
-class TransactionStore(BaseSlavedStore):
-    get_destination_retry_timings = TransactionStore.__dict__[
-        "get_destination_retry_timings"
-    ]
-    _get_destination_retry_timings = DataStore._get_destination_retry_timings.__func__
-    set_destination_retry_timings = DataStore.set_destination_retry_timings.__func__
-    _set_destination_retry_timings = DataStore._set_destination_retry_timings.__func__
-
-    prep_send_transaction = DataStore.prep_send_transaction.__func__
-    delivered_txn = DataStore.delivered_txn.__func__
+class TransactionStore(TransactionStore, BaseSlavedStore):
```

Comment:
oh god. `TransactionStore` inherits from `TransactionStore` ? I can see this is a pre-existing horridness, but ... ugh. Shouldn't this be a SlavedTransactionStore?

Reply:
[indeed if it can inherit the whole of `synapse.storage.transactions.TransactionStore`, can't it just be replaced with that?]

Reply:
Oh good god, I has completely missed that.

Reply:
I think we want to make sure they still inherit from BaseSlavedStore, though it might be simpler to just ensure that happens when we construct them in synapse/app/*. I'll leave it as a SlavedTransactionStore as we do that quite a bit for the other stores, and we can change it separately.